eng/Workarounds.targets

```diff
@@ -37,6 +37,28 @@
     </KnownFrameworkReference>
   </ItemGroup>
 
+<!-- Workaround while there is no 5.0 SDK available, copy known apphost/framework reference info from 3.0 -->
+  <ItemGroup>
+    <KnownAppHostPack 
+      Include="@(KnownAppHostPack->WithMetadataValue('TargetFramework', 'netcoreapp3.0'))"
+      TargetFramework="netcoreapp5.0"
+      />
+    <KnownFrameworkReference
+      Include="@(KnownFrameworkReference->WithMetadataValue('TargetFramework', 'netcoreapp3.0'))"
+      TargetFramework="netcoreapp5.0"
+      />
+  </ItemGroup>
+
+  <ItemGroup>
+    <!-- Reference base shared framework at incoming dependency flow version, not bundled sdk version. -->
```

Comment:
This is not a workaround and shouldn't go in this file.

For 3.0, this was already handled here:
https://github.com/aspnet/AspNetCore/blob/2e4274cb67c049055e321c18cc9e64562da52dcf/Directory.Build.targets#L114-L134

There are some more sophisticated rules around which versions to pick that should be carried forward.

One thing is that it is better to Update FrameworkReference as below than KnownFrameworkReference as there. KnownFrameworkReference is not meant to be mutated. It simply provides defaults that FrameworkReference can override.

Reply:
The thing is that this line and the line in D.B.targets are trying to do the same-ish thing. It should all be consolidated in one place, even if you choose to take a cleaner approach later. It is very confusing that 5.0 is handled here.

If DefaultNetCoreTargetFramework is v5.0, then I don't think this line is needed, though this line is closer to how I would want things to work, updating FrameworkReference, not KnownFrameworkReference.

Reply:
I've (attempted to) consolidate these 2 workarounds into a single `FrameworkReference` update in d.b.targets